### PR TITLE
Add GLSL linter using glslang

### LINKED ISSUE #592: Glsl linting
Any chance of getting a linter for GLSL?
Love this project, don't wanna have to boot up syntastic again...

I notice that the syntastic version is super simple, like 50 lines of code. Maybe I could do this, is it hard to port over these syntax checkers that are implemented in syntastic?

https://github.com/vim-syntastic/syntastic/blob/master/syntax_checkers/glsl/cgc.vim

Thanks

### PR DESCRIPTION
**Pull request by svenstaro**
Adds a GLSL linter using glslang. Also fixes #592.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,9 +91,10 @@ formatting.
 | Erlang | [erlc](http://erlang.org/doc/man/erlc.html), [SyntaxErl](https://github.com/ten0s/syntaxerl) |
 | Fortran | [gcc](https://gcc.gnu.org/) |
 | FusionScript | [fusion-lint](https://github.com/RyanSquared/fusionscript) |
+| GLSL | [glslang](https://github.com/KhronosGroup/glslang) |
 | Go | [gofmt](https://golang.org/cmd/gofmt/), [go vet](https://golang.org/cmd/vet/), [golint](https://godoc.org/github.com/golang/lint), [gometalinter](https://github.com/alecthomas/gometalinter) !!, [go build](https://golang.org/cmd/go/) !!, [gosimple](https://github.com/dominikh/go-tools/tree/master/cmd/gosimple), [staticcheck](https://github.com/dominikh/go-tools/tree/master/cmd/staticcheck) |
 | GraphQL | [gqlint](https://github.com/happylinks/gqlint) |
-| Haml | [haml-lint](https://github.com/brigade/haml-lint)
+| Haml | [haml-lint](https://github.com/brigade/haml-lint) |
```
Comment (svenstaro) [inline README.md]: Technically out of scope but I think this fixes a small markup problem.

File: doc/ale.txt [hunk 2]
```diff
@@ -46,6 +46,8 @@ CONTENTS                                                         *ale-contents*
       gcc.................................|ale-fortran-gcc|
     fusionscript..........................|ale-fuse-options|
       fusion-lint.........................|ale-fuse-fusionlint|
+    glsl..................................|ale-glsl-options|
+      glslang.............................|ale-glsl-glslang|
```
Comment (w0rp) [inline doc/ale.txt]: The Bash script pointed out that there's no tag for this doc tag link `|ale-glsl-glslang|`. I think the name is right here. You should make the heading name and the tag in `doc/ale-glsl.txt` match what you have here.

File: doc/ale-glsl.txt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  Since Vim does not detect the glsl file types out-of-the-box, you need the
+  runtime files for glsl from here: https://github.com/tikhomirov/vim-glsl
+
+  Note that the current glslang-based linter expects glslangValidator in
+  standard paths. If it's not installed system-wide you can set
+  |g:ale_glsl_glslang_executable| to a specific path.
+
+
+===============================================================================
+glslangValidator                                    *ale-glsl-glslangValidator*
```
Comment (w0rp) [inline doc/ale-glsl.txt]: I recommend changing these to `glslang` and `*ale-glsl-glslang*`. See my other comment.

File: doc/ale.txt (hunk 2 above)
Comment (svenstaro) [reply doc/ale.txt]: Oh yeah I was apparently moving these things around at some point. The name here is the one I intended.

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**